fuel/src/main/kotlin/com/github/kittinunf/fuel/toolbox/extensions/ForceMethod.kt

```diff
@@ -20,16 +20,20 @@ fun HttpURLConnection.forceMethod(method: Method) {
             } catch (ex: NoSuchFieldException) {
                 // ignore
             }
-            (arrayOf(this.javaClass.superclass, this.javaClass)).forEach {
-                try {
-                    it.getDeclaredField("method").apply {
-                        this.isAccessible = true
-                    }.set(this, method.value)
-                } catch (ex: NoSuchFieldException) {
-                    // ignore
-                }
-            }
+            this.forceMethod(this.javaClass, method)
         }
         else -> this.requestMethod = method.value
     }
+}
+
+private fun HttpURLConnection.forceMethod(clazz: Class<in HttpURLConnection>, method: Method) {
+    try {
+        clazz.getDeclaredField("method").apply {
+            this.isAccessible = true
+        }.set(this, method.value)
+    } catch (ex: NoSuchFieldException) {
+        if (HttpURLConnection::class.java.isAssignableFrom(clazz.superclass)) {
+            this.forceMethod(clazz.superclass, method)
```

Comment:
This is what the fix boils down to, right?

Reply:
That's right.

More specifically, by replacing the iteration : 

```kotlin
(arrayOf(this.javaClass.superclass, this.javaClass)).forEach {
```

with a new recursive method which traverses the hierarchy of parent classes until it finds the correct one.